crates/rpc/rpc/src/eth/api/block.rs

```diff
@@ -63,65 +63,69 @@ where
         &self,
         block_id: BlockId,
     ) -> EthResult<Option<Vec<AnyTransactionReceipt>>> {
-        let mut block_and_receipts = None;
-
-        if block_id.is_pending() {
-            block_and_receipts = self
-                .provider()
+        // Fetch block and receipts based on block_id
+        let block_and_receipts = if block_id.is_pending() {
+            self.provider()
                 .pending_block_and_receipts()?
-                .map(|(sb, receipts)| (sb, Arc::new(receipts)));
+                .map(|(sb, receipts)| (sb, Arc::new(receipts)))
         } else if let Some(block_hash) = self.provider().block_hash_for_id(block_id)? {
-            block_and_receipts = self.cache().get_block_and_receipts(block_hash).await?;
-        }
+            self.cache().get_block_and_receipts(block_hash).await?
+        } else {
+            None
+        };
+
+        // If no block and receipts found, return None
+        let (block, receipts) = match block_and_receipts {
+            Some(data) => data,
+            None => return Ok(None),
+        };
```

Comment:
this can be replaced with let Some() else {return Ok(None)}

Reply:
this looks even more clear, thanks